Add test for DNS function

CedricMoreau opened the pull request.

Commit:
Add test for DNS function

alagoutte commented on Tests/integration/DNS.Tests.ps1:
```diff
@@ -0,0 +1,42 @@
+
+Describe  "Set-ArubaSWDns" {
+    It "Set ArubaSWDns ip server 1" {
+        Set-ArubaSWDns -mode Manual -server1 10.44.1.1
```
use public DNS (Google, Cloudflare, Quad9) or private example network

alagoutte commented on Tests/integration/DNS.Tests.ps1:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    It "Set ArubaSWDns dns domain names" {
+        Set-ArubaSWDns -mode Manual -server1 10.44.1.1 -server2 8.8.8.8 -domain test,tttt
```
Set example domain name (example.org, example.com...)

alagoutte commented on Tests/integration/DNS.Tests.ps1:
```diff
@@ -0,0 +1,42 @@
+    It "Set ArubaSWDns dns domain names" {
+        Set-ArubaSWDns -mode Manual -server1 10.44.1.1 -server2 8.8.8.8 -domain test,tttt
+        $dns = Get-ArubaSWDns
+        $dns.dns_domain_names | Should be "test,tttt"
```
check also server1/server2 dns ?

alagoutte reviewed (changes requested):
Can you merge with #73 ?

alagoutte commented on Tests/integration/DNS.Tests.ps1:
```diff
@@ -0,0 +1,42 @@
+    It "Set ArubaSWDns ip server 1" {
+        Set-ArubaSWDns -mode Manual -server1 10.44.1.1
+        $dns = Get-ArubaSWDns 
+        $dns.server_1.octets | Should be "10.44.1.1"
```
check also the mode ?

alagoutte commented on Tests/integration/DNS.Tests.ps1:
```diff
@@ -0,0 +1,42 @@
+        Remove-ArubaSWDns -mode Manual -server1 none -server2 none 
+    }
+}
+
```
add a example with DHCP mode ?

alagoutte replied:
always valid request

Commit:
Fix after review #1

Commit:
Fix after review #2

alagoutte commented on Tests/integration/DNS.Tests.ps1:
```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+../common.ps1
```
missing a . before

alagoutte commented on Tests/integration/DNS.Tests.ps1:
```diff
@@ -0,0 +1,44 @@
+
+../common.ps1
+
+$mysecpassword = ConvertTo-SecureString $password -AsPlainText -Force
```
no needed.. (look other file)

alagoutte reviewed (commented):
need some fix...

alagoutte committed:
DNS(tests): Remvoe undeed include

alagoutte committed:
DNS(tests): fix common.ps1 include

missing dot

alagoutte committed:
DNS(tests): fix trailing whitespace

alagoutte committed:
DNS(tests): use 2 different DNS provider :-)

Cloudflare DNS (1.1.1.1) and Google DNS (8.8.8.8)

alagoutte committed:
DNS(tests): Add -noconfirm to Remove-ArubaSWDNS

alagoutte committed:
DNS(Tests): Add check when Set-ArubaDNS

Check the .version for DNS server and change also it is NullOrEmpty when not configured

alagoutte committed:
DNS(tests): Add Remove-ArubaSWDNS tests

alagoutte committed:
DNS(Tests): Add test with multiple domain names

alagoutte committed:
DNS(tests): Check also domain_name (when it is not set)

alagoutte committed:
DNS(tests): Remove trailing whitespace

alagoutte committed:
DNS(tests): Fix Missing BOM encoding for non-ASCII encoded file (found by PSSA)
